src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs

```diff
@@ -70,11 +70,7 @@ public void Dispose()
                 Task clientConnect = clientConnection.ConnectAsync(serverEndPoint.Address, serverEndPoint.Port);
                 Task<TcpClient> serverAccept = server.AcceptTcpClientAsync();
 
-                Assert.True(
-                    Task.WaitAll(
-                        new Task[] { clientConnect, serverAccept },
-                        TestConfiguration.PassingTestTimeoutMilliseconds),
-                    "Client/Server TCP Connect timed out.");
+                Assert.True(await TestConfiguration.WhenAllWithTimeout(clientConnect, serverAccept), "Client/Server TCP Connect timed out.");
```

Comment:
This is better than what was there before.  It would be even better to do a `WhenAllOrAnyFailedWithTimeout`, where it'll complete when either a) both tasks have completed, b) either of them has completed unsuccessfully, or c) the timeout occurred, whichever comes first.  That's helpful in situations where the two operations being waited on are related, such that if one fails, the other is likely to never complete, and it'd be better to fail fast and know why we failed rather than having to wait for the timeout and miss out on the underlying reason.  I don't believe we have such a helper yet, but either you could add one, or we could add it later.

Reply:
Yeah that seems reasonable, so it would return false for any failed as well? I can take a look at putting that in the test extensions for tasks that has the TimeoutAfter

Reply:
> so it would return false for any failed as well?

It wouldn't return false, it would propagate the failure.

Reply:
what about this?

https://github.com/dotnet/corefx/pull/25345/commits/21ecf922ababf57a8b58424ff26cc2b02ef8e5f0

The reason for leaving the return of true/false is just that there are a number of places that there are asserts and its using that to relay that there was a timeout. I am easy either way with it just throwing. 

I am also easy about it being called WithTimeout or just being with timeout based on it's method signature...

Reply:
Actually looking at it, I would just throw the exception in this method, and can wrap it inside the test helper for the networking stuff to keep the true/false.

Reply:
the TimeoutException that is.